### ISSUE TITLE
No Buffering state ever received, only Connecting and Paused

### ISSUE THREAD

**Issue by austinried** [doublesymmetry/react-native-track-player#1476]
**Describe the Bug**
Using `v2.2.0-rc3` I have noticed I never receive the Buffering state anymore.  I get a Connecting state when the queue first starts playing, and then a quick Paused state when skipping tracks/seeking within a track.  Here's an example log of states:

```
 LOG  None
 LOG  Connecting
 LOG  Playing
 LOG  Paused  <--- Skip to next track
 LOG  Playing
 LOG  Paused  <--- Seek within track
 LOG  Playing
```

**Steps To Reproduce**
Load up a queue of URLs, watch state transitions as you skip/seek tracks.

**Environment Info:**
Paste the results of `npx react-native info`
```
System:
    OS: Linux 5.4 Ubuntu 20.04.4 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    Memory: 158.45 MB / 15.39 GB
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 14.19.1 - /tmp/yarn--1650003558066-0.14211410378890887/node
    Yarn: 1.22.18 - /tmp/yarn--1650003558066-0.14211410378890887/yarn
    npm: 6.14.16 - ~/.nvm/versions/node/v14.19.1/bin/npm
    Watchman: 4.9.0 - /usr/bin/watchman
  SDKs:
    Android SDK:
      Android NDK: 22.1.7171670
  IDEs:
    Android Studio: Not Found
  Languages:
    Java: 11.0.12 - /home/austin/.sdkman/candidates/java/current/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.67.4 => 0.67.4 
  npmGlobalPackages:
    *react-native*: Not Found
```
Paste the exact `react-native-track-player` version you are using
> `2.2.0-rc3`

Real device? Or simulator?
> Two simulated devices (one running Android 12, the other Android 9) one real device (Pixel 4 Android 12)

What OS are you running?
> Ubuntu 20.04

**How I can Help**
What can you do to help resolve this?
> I can test suggested changes

Have you investigated the underlying JS or Swift/Android code causing this bug?
> No, I can try but I have very little Java experience and none in Kotlin

Can you create a Pull Request with a fix?
> Probably not

**Comment by jspizziri** [comment]
@mpivchev this looks like a relevant rewrite issue.

@austinried , if you're willing to take a crack at a fix I'm happy to try and provide guidance. It's a great way to learn new tech and level up 😜

**Comment by austinried** [comment]
Sure, I can poke around at stuff as good as the next person. Just want to set your expectations on what the quality of a Kotlin PR from me would be. 😅

Just in case it's related I will add some info about my build process to #1475 which I suspect could be involved there.

**Comment by jspizziri** [comment]
No judgement, just appreciation!

I'm sure it'll be great and that's what the remote prices is for anyways 😃

**Comment by mpivchev** [comment]
Did you receive the `Buffering` state in `rc1` and `rc2`?

**Comment by austinried** [comment]
@mpivchev looks like no, I just tested the example app in `rc1` and `rc2` and it's the same thing, no `Buffering` state.

**Comment by jspizziri** [comment]
I've done some investigation here and this is what I've found:

1. The `Buffering` event is in fact being fired in `KotlinAudio` as expected.
2. The issue seems to be a combination of 2 things:
    1. Events being emitted in rapid succession overriding one another.
    2. A potential issue with [how the `PAUSED` event is emitted.](https://github.com/doublesymmetry/KotlinAudio/blob/main/kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/BaseAudioPlayer.kt#L387)
    
When playing a track and then skipping to the next one a `Buffering` event is emitted and then immediately followed by a `Paused` event (fired by the [`onIsPlayingChanged`](https://github.com/doublesymmetry/KotlinAudio/blob/main/kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/BaseAudioPlayer.kt#L387)) immediately afterwards, resulting in only the `Paused` event being emitted.

My big question is, why are we even emitting a `Paused` event in this particular scenario, as no pause was actually caused. If we _do_ need to emit a paused event in this scenario then we have effectively a race-condition scenario. `Paused` should technically be emitted _before_ the `Buffering` state, but the buffering state is faster, and therefore getting swallowed.

**Comment by jspizziri** [comment]
The fix for this is in place and will be released in `v3.2`

**Comment by bradleyflood** [comment]
Here's `v3.2` https://github.com/doublesymmetry/react-native-track-player/releases/tag/v3.2.0

Should this be closed now?
